### PR TITLE
[RESTEASY-1719] changed processing for a Cookie as a String in a resp…

### PR DESCRIPTION
**Pull request by rsearls**
…onse object

### CODE REVIEW

File: resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java [hunk 1]
```diff
@@ -408,9 +409,16 @@ public static void commitHeaders(BuiltResponse jaxrsResponse, HttpResponse respo
                   response.addNewCookie(cookie);
                   it.remove();
                }
+               else if (next instanceof Object)
```
Comment (NicoNes) [inline resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java]: `next instance Object` will always be true right ?

Comment (rsearls) [reply resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java]: Sadly yes.  I am testing a change for this.

**Comment by rsearls** [comment]
Code change to check for char[] or String pushed

File: resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java [hunk 2]
```diff
@@ -402,15 +403,22 @@ public static void commitHeaders(BuiltResponse jaxrsResponse, HttpResponse respo
...
             {
                Object next = it.next();
+               String typeName = next.getClass().getTypeName();
                if (next instanceof NewCookie)
                {
                   NewCookie cookie = (NewCookie) next;
                   response.addNewCookie(cookie);
                   it.remove();
                }
+               else if ("java.lang.String".equals(typeName) || "char[]".equals(typeName))
```
Comment (NicoNes) [inline resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java]: Actually I think that it is not a good idea to only allow `NewCookie `, `String `or `char[] `as valid Set-Cookies header values.
Any object  should be considered as valid as long as there is an instance of `javax.ws.rs.ext.RuntimeDelegate.HeaderDelegate` able to convert it to a String or using it `toString() ` method.
It seems that such classes already exist in the project such as `org.jboss.resteasy.plugins.delegates.CookieHeaderDelegate` wich handle `javax.ws.rs.core.Cookie` or `org.jboss.resteasy.plugins.delegates.NewCookieHeaderDelegate` wich handle `javax.ws.rs.core.NewCookie`.

The java doc about `javax.ws.rs.core.Response.ResponseBuilder.header(String, Object)` is:

> 
>            Add an arbitrary header.
>           
>            @param name  the name of the header
>            @param value the value of the header, the header will be serialized
>                         using a {@link javax.ws.rs.ext.RuntimeDelegate.HeaderDelegate} if
>                         one is available via {@link javax.ws.rs.ext.RuntimeDelegate#createHeaderDelegate(java.lang.Class)}
>                         for the class of {@code value} or using its {@code toString} method
>                         if a header delegate is not available. If {@code value} is {@code null}
>                         then all current headers of the same name will be removed.
>            @return the updated response builder.

Comment (rsearls) [reply resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java]: javax.ws.rs.ext.RuntimeDelegate.HeaderDelegate  should not be a concern.
The call to response.getOutputHeaders().add in actuality calls
HttpServletResponseHeaders.add which in turn calls
HttpServletResponseHeaders.addResponseHeaders line 93
This method gets the proper RuntimeDelegate.HeaderDelegate and executes it.

**Comment by rsearls** [comment]
I've made the changes and added test cases

**Comment by NicoNes** [comment]
Hi @rsearls 

Thanks 👍

**Pull request merged**